Source/WebKit/WebProcess/WebPage/FindController.cpp

```diff
@@ -293,11 +289,23 @@ void FindController::findString(const String& string, OptionSet<FindOptions> opt
     }
 
     RefPtr<WebPage> protectedWebPage = m_webPage;
+
+#if ENABLE(IMAGE_ANALYSIS)
+    if (completionHandler) {
+        auto timer = makeUnique<PAL::HysteresisActivity>([protectedWebPage, string, options, maxMatchCount] (PAL::HysteresisState state) {
```

Comment:
I'd feel better if we captured the WebPage as a WeakPtr, to be 100% sure we don't have a retain-cycle:
WebPage -> Page -> ImageAnalysisQueue -> back to the WebPage via the HystererisActivity lambda.